### ISSUE TITLE
c-lightning should not disconnect when receiving a warning during quick close

### ISSUE THREAD

**Issue by t-bast** [ElementsProject/lightning#4777]
This is related to https://github.com/lightningnetwork/lightning-rfc/pull/904
Since c-lightning already supports warning messages, we should do this right now.

When c-lightning receives a warning during quick close (usually because the peer doesn't agree with the `fee_range` c-lightning sent), we really should not disconnect. If we stay connected and the node operator reads the warning message, they can reuse the `close` API to change the `fee_range` to something that the peer will agree with, and the mutual close is able to make progress.

But right now since c-lightning disconnects, on reconnection it will re-send the same `fee_range` automatically so it will instantly disconnect again, not giving the node operator an opportunity to update the `fee_range`.

**Comment by rustyrussell** [comment]
Yes, we should update if they try to close again... but not disconnecting would be weird?  We have nothing useful to say at this point.

**Comment by rustyrussell** [comment]
(We *always* close after a warning, for this reason)

**Comment by t-bast** [comment]
In that case you should allow the node operator to call the `close` API while disconnected, that would work as well indeed.

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: Mentioned in git commits (likely fixed); About deprecated feature

If you believe this issue should remain open, please comment and we can reopen it.
